### PR TITLE
Fix/pwa refresh

### PR DESCRIPTION
**Pull request by elvece**

### CODE REVIEW

File: frontend/projects/ui/src/app/components/toast-container/refresh-alert/refresh-alert.component.ts [hunk 1]
```diff
@@ -10,13 +11,23 @@ import { RefreshAlertService } from './refresh-alert.service'
...
   private readonly dismiss$ = new Subject<boolean>()
-
   readonly show$ = merge(this.dismiss$, this.refresh$)
 
   constructor(
     @Inject(RefreshAlertService) private readonly refresh$: Observable<boolean>,
+    private readonly updates: SwUpdate,
   ) {}
 
+  async pwaReload() {
```
Comment (MattDHill) [inline frontend/projects/ui/src/app/components/toast-container/refresh-alert/refresh-alert.component.ts:21]: Does this need a loading spinner? How long might it take?

Comment (elvece) [reply frontend/projects/ui/src/app/components/toast-container/refresh-alert/refresh-alert.component.ts:21]: Uncertain, need to test it live further. It is probably a good idea to have one. Will add.

**Comment by elvece** [comment]
@dr-bonez I need to test this with a real update, so please add this to the next 0344 OTA update (ideally for pi)

File: frontend/projects/ui/src/app/components/toast-container/refresh-alert/refresh-alert.component.html [hunk 2]
```diff
@@ -1,9 +1,32 @@
...
-  Your user interface is cached and out of date. Hard refresh the page to get
-  the latest UI.
-  <ul>
-    <li><b>On Mac</b>: cmd + shift + R</li>
-    <li><b>On Linux/Windows</b>: ctrl + shift + R</li>
-  </ul>
-  <a alertButton class="enter-click" role="cancel">Ok</a>
-</alert>
+<div *ngIf="!onPwa; else pwa">
+  <alert *ngIf="show$ | async" header="Refresh Needed" (dismiss)="onDismiss()">
```
Comment (MattDHill) [inline frontend/projects/ui/src/app/components/toast-container/refresh-alert/refresh-alert.component.html:2]: Could only do `show$ | async` once if you elevate in scope

**Comment by dr-bonez** [comment]
> @dr-bonez I need to test this with a real update, so please add this to the next 0344 OTA update (ideally for pi)

Do you? Why not just reflash and attach from another browser?

**Comment by elvece** [comment]
> > @dr-bonez I need to test this with a real update, so please add this to the next 0344 OTA update (ideally for pi)
> 
> Do you? Why not just reflash and attach from another browser?

I mean, I specifically want to test on iOS safari. I guess I could do Brave there since its the same under the hood...

File: frontend/projects/ui/src/app/components/toast-container/refresh-alert/refresh-alert.component.html (hunk 2 above)
Comment (elvece) [reply frontend/projects/ui/src/app/components/toast-container/refresh-alert/refresh-alert.component.html:2]: I actually cannot due to the way ionic styling is applied to the buttons and decided it was not worth it to reimplement myself.

**Review by MattDHill — DISMISSED** [review]

File: Makefile [hunk 3]
```diff
@@ -162,7 +162,7 @@ frontend/config.json: $(GIT_HASH_FILE) frontend/config-sample.json
 	npm --prefix frontend run-script build-config
 
 frontend/patchdb-ui-seed.json: frontend/package.json
-	jq '."ack-welcome" = $(shell yq '.version' frontend/package.json)' frontend/patchdb-ui-seed.json > ui-seed.tmp
+	jq '."ack-welcome" = "$(shell yq '.version' frontend/package.json)"' frontend/patchdb-ui-seed.json > ui-seed.tmp
```
Comment (elvece) [inline Makefile:165]: @dr-bonez do you see a problem adding those quotes for a linux env? I needed them for osx when running `make frontends`

**Pull request merged**